core/dogu_v2.go

```diff
@@ -0,0 +1,490 @@
+package core
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"strings"
+
+	"encoding/json"
+	"github.com/gosuri/uitable"
+)
+
+// GetLogger is an alias function to provide a different logger for the core.
+var GetLogger = func() logrus.FieldLogger { return logrus.StandardLogger() }
+
+// Volume is the volume struct of a dogu and will be used to define docker
+// volumes
+type Volume struct {
+	Name        string
+	Path        string
+	Owner       string
+	Group       string
+	NeedsBackup bool
+}
+
+// UnmarshalJSON sets the default value for NeedsBackup. We are preventing an infinite loop by using a local Alias type
+// to call json.Unmarshal again
+func (v *Volume) UnmarshalJSON(data []byte) error {
+	type Alias Volume
+	a := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(v),
+	}
+	v.NeedsBackup = true
+	if err := json.Unmarshal(data, &a); err != nil {
+		return err
+	}
+	return nil
+}
+
+// HealthCheck struct will be used to do readyness and health checks for the
+// final container
+type HealthCheck struct {
+	Type       string            // health check type tcp or state
+	State      string            // expected state for state health check, default is ready
+	Port       int               // port for tcp state check
+	Path       string            // path for http check
+	Parameters map[string]string // key value pairs for check specific parameters
+}
+
+// ExposedPort struct is used to define ports which are exported to the host
+type ExposedPort struct {
+	// Type contains the protocol type over which the container communicates (f. i. 'tcp').
+	Type string
+	// Container contains the mapped port on side of the container.
+	Container int
+	// Host contains the mapped port on side of the host.
+	Host int
+}
+
+// GetType returns type of expose port, the default is tcp
+func (ep *ExposedPort) GetType() string {
+	if ep.Type == "" {
+		return "tcp"
+	}
+	return ep.Type
+}
+
+// ExposedCommand struct represents a command which can be executed inside the
+// dogu
+type ExposedCommand struct {
+	Name        string
+	Description string
+	Command     string
+}
+
+// Names for ExposedCommands correspond with actual dogu descriptor instance values. Do not change because these come
+// with side effects.
+const (
+	ExposedCommandServiceAccountCreate = "service-account-create"
+	ExposedCommandServiceAccountRemove = "service-account-remove"
+	ExposedCommandBackupConsumer       = "backup-consumer"
+	ExposedCommandPreBackup            = "pre-backup"
+	ExposedCommandPostBackup           = "post-backup"
+	ExposedCommandPostUpgrade          = "post-upgrade"
+	ExposedCommandPreUpgrade           = "pre-upgrade"
+	ExposedCommandUpgradeNotification  = "upgrade-notification"
+)
+
+// EnvironmentVariable struct represents custom parameters that can change
+// the behaviour of a dogu build process
+type EnvironmentVariable struct {
+	Key   string
+	Value string
+}
+
+// EnvironmentVariable toString method
```

Comment:
This comment does not comply with the general golang doc comment format.
```suggestion
// String returns a string representation of this EnvironmentVariable
```

- [ ] done